internal/plog/level_test.go

```diff
@@ -4,14 +4,169 @@
 package plog
 
 import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"reflect"
 	"strconv"
 	"testing"
+	"time"
+	"unsafe"
 
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 	"k8s.io/component-base/logs"
 	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+	clocktesting "k8s.io/utils/clock/testing"
 )
 
+func TestFormat(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	var buf bytes.Buffer
+	ws := zapcore.Lock(zapcore.AddSync(&buf))
+
+	scanner := bufio.NewScanner(&buf)
+
+	now, err := time.Parse(time.RFC3339Nano, "2022-11-21T23:37:26.953313Z")
+	require.NoError(t, err)
+	fakeClock := clocktesting.NewFakeClock(now)
+
+	err = ValidateAndSetLogLevelAndFormatGlobally(ctx, LogSpec{Level: LevelDebug},
+		zap.WithClock(&clockAdapter{clock: fakeClock}),
+		zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+			// this is some dark magic to set a private field
+			outField := reflect.ValueOf(core).Elem().FieldByName("out")
+			outPointer := (*zapcore.WriteSyncer)(unsafe.Pointer(outField.UnsafeAddr()))
+			*outPointer = ws
+			return core
+		}),
+	)
+	require.NoError(t, err)
+
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+
+	const startLogLine = 55 // make this match the current line number
+
+	Info("hello", "happy", "day", "duration", time.Hour+time.Minute)
+	require.True(t, scanner.Scan())
+	require.NoError(t, scanner.Err())
+	require.JSONEq(t, fmt.Sprintf(`
+{
+  "level": "info",
+  "ts": "2022-11-21T23:37:26.953313Z",
+  "caller": "%s/level_test.go:%d$plog.TestFormat",
+  "msg": "hello",
+  "happy": "day",
+  "duration": "1h1m0s"
+}`, wd, startLogLine+2), scanner.Text())
+
+	Logr().WithName("burrito").Error(errInvalidLogLevel, "wee", "a", "b")
+	require.True(t, scanner.Scan())
+	require.NoError(t, scanner.Err())
+	require.JSONEq(t, fmt.Sprintf(`
+{
+  "level": "error",
+  "ts": "2022-11-21T23:37:26.953313Z",
+  "caller": "%s/level_test.go:%d$plog.TestFormat",
+  "msg": "wee",
+  "a": "b",
+  "error": "invalid log level, valid choices are the empty string, info, debug, trace and all",
+  "logger": "burrito"
+}`, wd, startLogLine+2+13), scanner.Text())
```

Comment:
This shows how the JSON logs will look (though they will be on a single line instead of multiple).  I think the `caller` information is really useful and overall I find this format to be very readable.